### DISCUSSION TITLE
Freeflyer Mass matrix and non-linear forces in WORLD_ALIGNED frame

### DISCUSSION THREAD

**Discussion by MaximilianDio** [stack-of-tasks/pinocchio#1904] [Q&A] [answered]
Hello,

I wonder if I can get the dynamic quantities for a model (single link) with a freeflyer root (Mass matrix and nonlinear effects) from crba and rnea represented in the world frame. If I use the functions in the standard way, I always get the quantities in the body frame. Of course, I could transform them but if there are already implemented versions I would prefer to use them. 

kind regards,
Max

**Comment by jcarpent** [comment, marked as answer]
According to Featherstone's book, because we work with the Lagrangian dynamics, we are directly in the LOCAL frames to get the best efficiency.
Having them in the WORLD frame is indeed possible and easy to do. You just need to rotate the quantities.
